Flaky unit tests showing up on windows runner

Strange, windows unit test took this in CI, but the other unit test runs worked:

```
com.ichi2.anki.DeckPickerTest > limitAppliedAfterReview FAILED

    java.lang.UnsupportedOperationException
        at com.ichi2.libanki.sched.AbstractDeckTreeNode.getNewCount(AbstractDeckTreeNode.java:176)
        at com.ichi2.anki.DeckPickerTest.lambda$limitAppliedAfterReview$6(DeckPickerTest.java:180)
        at androidx.test.core.app.ActivityScenario.lambda$onActivity$2$ActivityScenario(ActivityScenario.java:660)
        at androidx.test.core.app.ActivityScenario$$Lambda$4.run(Unknown Source)
        at androidx.test.core.app.ActivityScenario.onActivity(ActivityScenario.java:670)
        at com.ichi2.anki.DeckPickerTest.limitAppliedAfterReview(DeckPickerTest.java:178)
```

Likewise on my fork (plus in main repo for i18n sync PR) windows unit test took this, while other unit tests passed:

```
com.ichi2.anki.DeckPickerTest > confirmDeckDeletionDeletesEmptyDeck FAILED

    java.lang.Exception: Main looper has queued unexecuted runnables. This might be the cause of the test failure. You might need a shadowOf(getMainLooper()).idle() call.
        at org.robolectric.android.internal.AndroidTestEnvironment.checkStateAfterTestFailure(AndroidTestEnvironment.java:502)
        at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:581)
        at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:263)
        at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

        Caused by:
        java.lang.AssertionError: deck was deleted

        Expected: is <1>
             but: was <2>
            at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
            at com.ichi2.anki.DeckPickerTest.confirmDeckDeletionDeletesEmptyDeck(DeckPickerTest.java:197)
```

_Originally posted by @mikehardy in https://github.com/ankidroid/Anki-Android/pull/7652#issuecomment-725568056_

--- a/AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.java
+++ b/AnkiDroid/src/test/java/com/ichi2/anki/DeckPickerTest.java
@@ -192,7 +192,7 @@ public class DeckPickerTest extends RobolectricTest {
 
         deckPicker.confirmDeckDeletion(did);
 
-        advanceRobolectricLooper();
+        advanceRobolectricLooperWithSleep();
 
         assertThat("deck was deleted", getCol().getDecks().count(), is(1));
     }

